1_fetch/src/fetch_precip_nc.R

```diff
@@ -20,10 +20,12 @@ fetch_precip_data <- function(ind_file, view_polygon, times) {
   stencil_pts <- as.stencil_pt('xmin', 'ymin', 'low_left') %>% cbind(stencil_pts)
 
   # break times into chunks to avoid WAF rejections of queries that are "too big"
-  dateseq <- unique(c(
-    seq(as.POSIXct(times$start, tz='UTC'), as.POSIXct(times$end, tz='UTC'), by=as.difftime(3, units='days')),
-    as.POSIXct(times$end, tz='UTC')
-  ))
+  try_formats = c("%Y-%m-%d %H:%M:%OS", "%Y/%m/%d %H:%M:%OS", "%Y-%m-%d %H:%M", "%Y/%m/%d %H:%M", "%Y-%m-%d", "%Y/%m/%d")
+  dateformat <- try_formats[sapply(try_formats, function(fmt) all(!is.na(as.POSIXct(unlist(times), format=fmt))))]
```

Comment:
is this related to fixing the issue with `c()` or a new feature??

Reply:
Related. Rather than trying to get rid of the `c()`, which was introducing new headaches, I decided it was simpler to apply the c() to strings rather than POSIXct objects, and then parse them back into POSIXct after the concatenation. But to do that I needed to have the format written out, and I know we discussed in slack that it might be fine to assume the format of the dates is always '%Y-%m-%d', but by the time we had had that conversation I had figured out this approach which would allow us to specify hours if we wanted. In the end, if the format of `dates` is always like `2018-08-23`, then `dateformat` should always equal `%Y-%m-%d`.